packages/babel-parser/src/tokenizer/index.js

```diff
@@ -1197,7 +1197,7 @@ export default class Tokenizer extends ParserErrors {
       const code = this.input.charCodeAt(this.state.pos);
       let val;
 
-      if (code === charCodes.underscore) {
+      if (code === charCodes.underscore && allowNumSeparator !== "bail") {
         const prev = this.input.charCodeAt(this.state.pos - 1);
         const next = this.input.charCodeAt(this.state.pos + 1);
         if (allowedSiblings.indexOf(next) === -1) {
```

Comment:
Not related to this PR: these errors should be thrown only when `allowNumSeparator` is `true`, otherwise the errors might be confusing.

Reply:
I'll open a PR when this is merged.